Adds filter to tags to be collected

sauljabin opened the pull request:
When an image_id has multiple tags, it's posible delete it, even when it should not.
Now, the garbage collector verifies if exists another tag that does not match.
For example: the image id 7602d1365872f9bf8b27bb7947f04dab91ad6cbd364457a206d1be4a4c535d32
has tags: build-646783 (to be delete) and release-34354678
build-646783 match with the regex, but release-34354678 doesn't.

Signed-off-by: Saúl Piña <sauljabin@gmail.com>

mssola commented:
@sauljabin thanks for this PR! Just so I can test this code, could you push an image out there so I can pull it and test this code locally? (or maybe show me an example image in which this can be reproduced). Thanks!

sauljabin commented:
Hi @mssola, thank you, for testing my approach. I was using the official docker-compose and a volume to do this, for example:

```yml
...
background:
.....
volumes:
      - ./secrets:/certificates:ro
      - ./garbage_collector.rb:/srv/Portus/lib/portus/background/garbage_collector.rb:ro

```

sauljabin commented:
@mssola, Additionally, I set this envs for background portus:

```
      - PORTUS_BACKGROUND=true
      - PORTUS_DELETE_ENABLED=true
      - PORTUS_DELETE_GARBAGE_COLLECTOR_ENABLED=true
      - PORTUS_DELETE_GARBAGE_COLLECTOR_OLDER_THAN=10
      - PORTUS_DELETE_GARBAGE_COLLECTOR_TAG=^(?!latest|release|stable).*
```

Of course you can put any tag.

sauljabin commented:
Sorry, @mssola i just used portus:2.4 running with a volume garbage_collector.rb. :sweat_smile:

vitoravelino commented:
@sauljabin I guess what @mssola wanted to know is an use case step-by-step for the change you are proposing.

sauljabin commented:
Oh sorry, jeje, well i was using images like: ubuntu or nginx, then i added a few tags.

For example: 

Case 1: (do not delete)
docker tag nginx:1.15 registry:5000/nginx:1.15
docker push registry:5000/nginx:1.15

docker tag nginx:1.15 registry:5000/nginx:latest
docker push registry:5000/nginx:latest

Case 2: (delete)
docker tag nginx:1.15 registry:5000/nginx:1.15
docker push registry:5000/nginx:1.15

Case 3: (do not delete)
docker tag nginx:1.15 registry:5000/nginx:latest
docker push registry:5000/nginx:latest

Where registry is pointing to 127.0.0.1 .

You can use any image. The thing is that one image_id have two tags, one to delete and another do not.

After push, i modified the create and update_at fields from de DB, to simulate the "older than" configuration. For example: 2018-01-01 .... (a year ago).

Tag pattern: ^(?!latest|release|stable).*
